### DISCUSSION TITLE
How do I migrate complete data

### DISCUSSION THREAD

**Discussion by ZiXia1** [plausible/analytics#3281] [Self-Hosted Support] [answered]
There is no detailed tutorial on migrating data in docs
Not very friendly to people like me who don't know much about docker and clickhouse

**Comment by ruslandoga** [comment]
👋 @ZiXia1 

Please check out `v2` upgrade instructions in https://github.com/plausible/analytics/discussions/3132

**Reply by ZiXia1** [reply]
I've seen this update description before, and my idea at the time was to migrate the data to a second server.

It would be nice to have the ability to export and import statistics files on a Web page

**Reply by ruslandoga** [reply]
Ah, I see. Sorry, I misunderstood. Right now we don't have guides for exporting and importing raw data between servers.

**Reply by ruslandoga** [reply, marked as answer]
The closest we have is https://github.com/plausible/hosting/blob/master/upgrade/postgres.md, which can be used to backup PostgreSQL data on one server and import it on another and an incomplete https://github.com/plausible/hosting/pull/66

**Comment by visini** [comment]
For posterity: If you're looking for a step-by-step guide on migrating complete data between servers, I recently wrote a blog post about it: https://camillovisini.com/coding/migrating-plausible-analytics

**Reply by ruslandoga** [reply]
👋 @visini 

Thank you for the write-up and for posting it here!

One nitpick:

> Next, we need to create a dump of the ClickHouse data. ClickHouse stores the actual analytics data, such as page views and unique visitors. The relevant tables are ingest_counters, sessions_v2, and events_v2.

`imported_*` tables might be relevant to some users too, especially if they migrated from Universal Analytics since UA no longer allows exports.

**Reply by visini** [reply]
Thank you @ruslandoga, good point. I've added a corresponding note 😊
